### PR TITLE
Done

### PR DESCRIPTION
**Pull request by HarlemSquirrel**

### CODE REVIEW

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,10 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "\t X O X O X O X O X O X O X O X O X O X O X"
+puts "\t\t Welcome to Tic Tac Toe!"
+puts "\t X O X O X O X O X O X O X O X O X O X O X"
+
+game = Game.new
+game.start
```
Comment (randallreedjr) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

Furthermore, by splitting the interface (`TicTacToeCLI`) from the implementation (`Game`) of the game allows for multiple interfaces (e.g. a web app) without requiring changes to the game code.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    position(input) == "X" || position(input) == "O"
+  end
+
+  def valid_move?(input)
+    # ensure that moves are between 1-9 and not taken
+    input.to_i.between?(1,9) && taken?(input) == false
+  end
+
+  def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
```
Comment (randallreedjr) [inline lib/board.rb]: Another way to instantiate an Array with `i` values is `Array.new(9, " ")` (see [Array.new](http://ruby-doc.org/core-2.2.0/Array.html#method-c-new)), where `9` is the number of elements in the array and `" "` is the default value for the elements. It's more concise and will prevent errors from repeatedly typing `" "`.

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  ]
+
+  def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    return @player_1 if @board.turn_count % 2 == 0
```
Comment (randallreedjr) [inline lib/game.rb]: Use [`Integer#even?`](http://ruby-doc.org/core-1.8.7/Integer.html#method-i-even-3F) or [`Integer#odd?`](http://ruby-doc.org/core-1.8.7/Integer.html#method-i-odd-3F)

`return @player_1 if @board.turn_count.even? `

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      tokens = [@board.cells[combo[0]], @board.cells[combo[1]], @board.cells[combo[2]]]
+      return "X" if tokens.count("X") == 3
+      return "O" if tokens.count("O") == 3
+    end
+    nil
+  end
+
+  def won?
+    return true if winner != nil
+    false
```
Comment (randallreedjr) [inline lib/game.rb]: You seem fond of guard clauses, but this one seems unnecessary. Generally speaking, returning explicit `true` and `false` values with an `if` condition is extraneous. Consider this alternative:
```
def won?
  winner != nil
end
```
which could be further simplified to
```
def won?
  winner
end
```
You could also use `!!winner` if you wanted to guarantee a boolean return value.

**Comment by randallreedjr** [comment]
@HarlemSquirrel Great job! I left you a few comments. Closing this PR for now, but feel free to re-open and respond if you would like to discuss any of them.

File: bin/tictactoe (hunk 1 above)
Comment (HarlemSquirrel) [inline bin/tictactoe]: Ah, I see! That makes a lot of sense. Thank you!

File: lib/board.rb (hunk 2 above)
Comment (HarlemSquirrel) [inline lib/board.rb]: Wow, that is cool! A very nice way to do this.

File: lib/game.rb (hunk 3 above)
Comment (HarlemSquirrel) [inline lib/game.rb]: That is definitely cleaner.

File: lib/game.rb (hunk 4 above)
Comment (HarlemSquirrel) [inline lib/game.rb]: I was having some issues at one point and lost faith in my logic for a moment. This is better.

**Comment by HarlemSquirrel** [comment]
Thank you for all the feedback! I learned a bunch!

**Pull request closed without merging**